### PR TITLE
Split sandbox code into separate packages

### PR DESCRIPTION
**Pull request by rautenrieth-da**
This PR splits `//ledger/sandbox` into 4 packages:
- `//ledger/daml-integration-api`: public code for ledger integrations. E.g., index, indexer, ledger API server.
- `//ledger/sandbox-common`: code common to sandbox classic and sandbox next. E.g., sandbox command line options.
- `//ledger/sandbox`: sandbox classic
- `//ledger/sandbox-next`: sandbox next

Additionally, it changes some folder names from `**/com/digitalasset/**` to `**/com/daml/**`, to correspond the the scala package names.

### CODE REVIEW

**Comment by S11001001** [comment]
Please take `com/daml` out of the filesystem path entirely; this is what we've done for other reorganizations.

**Comment by gerolf-da** [comment]
Thanks @rautenrieth-da. What do you think of the following changes?
* `sandbox` -> `sandbox-classic` (we'll probably rename this to daml-on-sql relatively soon, but it makes sense to use sandbox-classic for now)
* `sandbox-next` -> `sandbox`
* `daml-integration-api` -> `participant-integration-api`: using `participant` instead of the more generic `daml` is more expressive as to what this is about.

File: ledger/sandbox-classic/.gitattributes [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Enforce Unix newlines
```
Comment (stefanobaghino-da) [inline ledger/sandbox-classic/.gitattributes:1]: Why? Is this going to affect the Windows build?

Comment (rautenrieth-da) [reply ledger/sandbox-classic/.gitattributes:1]: Good point, this file is currently present in `//ledger/sandbox` and got copied to the new `sandbox-classic` when I split it. But all SQL files moved to `//ledger/participant-integration-api`, so this file should be deleted.

**Comment by gerolf-da** [comment]
Capturing ToDos together with @rautenrieth-da:

* Add `participant-integration-api` to `release/artifacts.yaml`
* Rename maven target artifact id of `sandbox-classic` from `sandbox` to `sandbox-classic`. complete maven coords: `com.daml:sandbox-classic`. 
* Add `sandbox-common` to `release/artifacts.yaml`
* Change `sandbox-binary_deploy` to `sandbox-classic-binary_deploy` in https://github.com/digital-asset/daml/blob/ae65f93e01525351f7b7ac325811ee359dbdbdbd/ci/build-unix.yml#L89-L90
* use `//ledger/sandbox/sandbox-binary` for the following targets (if no scenarios are used):
  * `//language-support/hs/bindings/BUILD.bazel`
  * `//language-support/ts/codegen/BUILD.bazel`
  * `//language-support/scala/codegen-sample-app/BUILD.bazel`
  * `//triggers/service/daml-platform.sh`
  * `//daml-script/test/BUILD.bazel`
  * `//daml-assistant/daml-helper/BUILD.bazel`
  * `//libs-haskell/test-utils/DA/Test/Sandbox.hs`
  * `//compiler/damlc/tests/BUILD.bazel`

**Comment by gerolf-da** [comment]
@S11001001: are you referring to json api? normally we follow the folder-per-package convention.

**Comment by S11001001** [comment]
@gerolf-da I'm referring to the discussions around #5343, in which the conclusion of the discussion, as I recall, was that if we _did_ change the paths, we would simply remove that segment rather than replacing it. We ultimately did not change the paths as part of #5343, but that doesn't alter any of the benefits of simply removing the paths or drawbacks of replacing them.

**Pull request merged**